### PR TITLE
Graphics_oM: Gradient object

### LINKED ISSUE #626: Graphics_oM: Gradient object
#### Description:
A `SortedDictionary` with keys (`decimal`) and values (`Color`), which would have engine methods to get a `Color` for any value between 0 to 1

Would enable more freedom in displaying `RenderMesh` and others in the future.

### PR DESCRIPTION
**Pull request by kThorsager**
### Issues addressed by this PR

Closes #626 
A SortedDictionary with keys (decimal) and values (Color), which would have engine methods to get a Color for any value between 0 to 1

### CODE REVIEW

**Comment by kThorsager** [comment]
Noticed that `IObjects` did not work as datasets (even if I had added a name) and have hence changed it from a `IObject` to a `IBHoMObject`, perhaps compatibility for `IObject` as datasets is something to look into for when smaller data like this is to be added

File: Graphics_oM/Colors/Gradient.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2018, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Graphics_oM/Colors/Gradient.cs]: Copyright should be `2019` not `2018`

File: Graphics_oM/Colors/Gradient.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+namespace BH.oM.Graphics
+{
+    public class Gradient : BHoMObject
+    {
+        /***************************************************/
+        /****            Public Properties              ****/
+        /***************************************************/
+
+        [Description("A List of Tuples containing a Color and double. The double signifies the position within the gradient and should be between 0 to 1")]
+        public SortedDictionary<decimal,Color> Markers { get; set; } = new SortedDictionary<decimal, Color>();
```
Comment (FraserGreenroyd) [inline Graphics_oM/Colors/Gradient.cs]: Put a space after the `,` in `public SortedDictionary<decimal, Color>` to be consistent with style guide.

**Comment by kThorsager** [comment]
@FraserGreenroyd as discussed to review

**Comment by kThorsager** [comment]
@rwemay Good catch, updated it now

**Pull request merged**